packages/plugin-core/src/components/lookup/utils.ts

```diff
@@ -390,16 +407,180 @@ export class PickerUtilsV2 {
     return false;
   }
 
-  static promptVault(overrides?: DVault[]): Promise<DVault | undefined> {
-    const vaults = overrides || DendronWorkspace.instance().vaultsv4;
+  public static async getOrPromptVaultForNewNote({
+    vault,
+    fname,
+    autoSuggest = true,
+  }: {
+    vault: DVault;
+    fname: string;
+    autoSuggest?: boolean;
+  }): Promise<DVault | undefined> {
+    const vaultSuggestions = await PickerUtilsV2.getVaultRecommendations({
+      vault,
+      fname,
+      autoSuggest,
+    });
+
+    if (vaultSuggestions?.length === 1) {
+      return vaultSuggestions[0].vault;
+    }
+
+    return PickerUtilsV2.promptVault(vaultSuggestions);
+  }
+
+  public static promptVault(overrides?: DVault[]): Promise<DVault | undefined>;
+  public static promptVault(
+    overrides?: VaultPickerItem[]
+  ): Promise<DVault | undefined>;
+  public static async promptVault(
+    overrides?: VaultPickerItem[] | DVault[]
+  ): Promise<DVault | undefined> {
+    const pickerOverrides = isDVaultArray(overrides)
+      ? overrides.map((value) => {
+          return { vault: value };
+        })
+      : overrides;
+
+    const vaults: VaultPickerItem[] =
+      pickerOverrides ??
+      DendronWorkspace.instance().vaultsv4.map((value) => {
+        return { vault: value };
+      });
+
     const items = vaults.map((ent) => ({
       ...ent,
-      label: ent.fsPath,
+      label: ent.label ? ent.label : ent.vault.fsPath,
     }));
-    const resp = VSCodeUtils.showQuickPick(items) as Promise<
-      DVault | undefined
-    >;
-    return resp;
+    const resp = await VSCodeUtils.showQuickPick(items, {
+      title: "Select Vault",
+    });
+
+    return resp ? resp.vault : undefined;
+  }
+
+  // Determine which are the most appropriate vault(s) to create this note in.
+  // Vaults determined as better matches appear earlier in the returned array
+  static async getVaultRecommendations({
+    vault,
+    fname,
+    autoSuggest = true,
+  }: {
+    vault: DVault;
+    fname: string;
+    autoSuggest?: boolean;
+  }):
+  Promise<VaultPickerItem[] | undefined> {
+    let vaultSuggestions: VaultPickerItem[] = [];
+
+    const engine = getWS().getEngine();
+
+    // Only 1 vault, no other options to choose from:
+    if (engine.vaults.length <= 1) {
+      return Array.of({ vault });
+    }
+
+    const hierarchies = fname.split(".").slice(0, -1);
```

Comment:
we refer to the top part of a hierarchy as the `domain`. there's actually a utility method to get it [here](https://github.com/dendronhq/dendron/blob/feat/multi-vault-note-creation/packages/common-all/src/dnode.ts#L243:L243). its not used anywhere so you are free to use it.

it currently throws an error if the node you pass in is the root node but feel free to change it to return the root instead

Reply:
I see - I've diverged a bit here though; whereas the `DNodeUtils` uses `DNodeProps` and `DNodePropsDict`, I stripped down the args on this method to just have fname of the proposed new note.